cni/pkg/nodeagent/cni-watcher.go

```diff
@@ -160,7 +160,7 @@ func (s *CniPluginServer) ReconcileCNIAddEvent(ctx context.Context, addCmd CNIPl
 	var ambientPod *corev1.Pod
 	var err error
 	// The plugin already consulted the k8s API - but on this end handler caches may be stale, so retry a few times if we get no pod.
-	for ambientPod, err = s.handlers.GetPodIfAmbient(addCmd.PodName, addCmd.PodNamespace); (ambientPod == nil) || (retries < maxStaleRetries); retries++ {
+	for ambientPod, err = s.handlers.GetPodIfAmbient(addCmd.PodName, addCmd.PodNamespace); (ambientPod == nil) && (retries < maxStaleRetries); retries++ {
```

Comment:
Didn't see this code before. Are we only waiting for 5ms? That seems pretty short given we are talking about pushing an event from etcd to the api server to cni

Reply:
I'm not sure what the upper and lower bounds of the latency typically are - erred on the side of short + logs.

I can make it 10ms x 10

Reply:
I don't want to do something like an exponential backoff here, or wait forever tho - we are blocking pod scheduling by delaying here so keeping it short is important.